docs/AUTH_PROTOCOL.md

```diff
@@ -0,0 +1,393 @@
+    - Arguments:
+        - `feedback`: The feedback to provide to the user. Valid values follow:
+            - `scanning`: The facial recognition system sees a face and is attempting to scan it
+            - `moveCloser`: The user needs to move closer to the camera
+            - `moveFurther`: The user needs to move further from the camera
+            - `recenter`: The user needs to ensure that they're facing and centered on the camera
+            - `blink`: Ask the user to blink, as a liveness check
+            - `lost`: The facial recognition system has lost sight of the user's face
+            - Unknown values should be treated as equivalent to `scanning`
+    - Response: none
+
+- `faceFailed`: The facial recognition system has scanned the face and determined that it does not match
+    - Sent from authenticator to frontend
+    - This is sent only while the `face` mechanism is secondary. If it's primary, `authenticationFailed` will be sent instead
+    - Arguments:
+        - `retrying`: If true, the facial recognition system will keep trying to scan the next face it sees and the `face` mechanism remains active. Otherwise, facial recognition is disabled
+    - Response: none
+
+#### Passkey events
+
+These events might be emitted while the `passkey` or `monitorPasskey` mechanisms are active:
+
+- `passkeyInserted`: A FIDO security token has been inserted
+    - Sent from frontend to authenticator
+    - Arguments: none
+    - Response: the authenticator should start the `passkey` mechanism
+        - If the authenticator has a flow dedicated to passkeys, then it should also emit the `flowChanged` event to notify the frontend that we've entered the passkey flow
+
+- `passkeyRemoved`: The FIDO security token has been removed
+    - Sent from frontend to authenticator
+    - Arguments: none
+    - Response: the authenticator can start the `monitorPasskey` mechanism:
+        - If the authenticator has a flow dedicated to passkeys and `flowChanged` has been emitted, the `monitorPasskey` mechanism should be the primary mechanism
+        - Otherwise, the `monitorPasskey` mechanism should be secondary, and the authenticator should pick a different appropriate primary mechanism
+
+#### Smart card events
+
+These events might be emitted while the `smartcard` or `monitorSmartcard` mechanisms are active:
+
+- `smartcardInserted`: A smart card has been inserted
+    - Sent from frontend to authenticator
+    - Arguments: none
+    - Response: the authenticator should read the smart card for a list of certificates, then start the `smartcard` mechanism
+        - If the authenticator has a flow dedicated to smart cards, then it should also emit the `flowChanged` event to notify the frontend that we've entered the smart card flow
+
+- `smartcardRemoved`: The smart card has been removed
+    - Sent from frontend to authenticator
+    - Arguments: none
+    - Response: the authenticator can start the `monitorSmartcard` mechanism:
```

Comment:
the mess starts when you want to make support for mulitple smart cards. You need to monitor new smart cards even if the smart card is already inserted (because the user might want to authenticate using different smart card that they plugged in later).